tfhe/src/shortint/ciphertext/zk.rs

```diff
@@ -90,10 +90,57 @@ impl ProvenCompactCiphertextList {
         &self,
         casting_mode: ShortintCompactCiphertextListCastingMode<'_>,
     ) -> crate::Result<Vec<Ciphertext>> {
+        let per_list_casting_mode: Vec<_> = match casting_mode {
+            ShortintCompactCiphertextListCastingMode::CastIfNecessary {
+                casting_key,
+                functions,
+            } => match functions {
+                Some(functions) => {
+                    let functions_count = functions.len();
+                    let required_functions_count: usize = self
+                        .proved_lists
+                        .iter()
+                        .map(|list| list.0.ct_list.lwe_ciphertext_count().0)
+                        .sum();
```

Comment:
I'm confused is it really one fn per ciphertext and not the possibility to run many functions for one ct ?

Reply:
sorry there are two levels of arrays

first one is per block, and can contain many functions

Reply:
this checks the first level, I'll rename to make things clearer